### PR TITLE
Add MXCLABEField for Mexico

### PR DESCRIPTION
**Pull request by agustin380**
This pull request adds a custom field for Clave Bancaria Estandarizada, used in all financial transactions in Mexico.

More info (in spanish):
https://es.wikipedia.org/wiki/CLABE

### CODE REVIEW

File: localflavor/mx/forms.py [hunk 1]
```diff
@@ -163,6 +163,50 @@ def _has_inconvenient_word(self, rfc):
...
 
 
+class MXCLABEField(RegexField):
+    """This field validates a CLABE (Clave Bancaria Estandarizada).
+
+    A CLABE is a 18-digits long number. The first 6 digits denote bank and branch number.
+    The remaining 12 digits denote an account number, plus a verifying digit.
+
+    More info:
+    https://es.wikipedia.org/wiki/CLABE
```
Comment (benkonrath) [inline localflavor/mx/forms.py]: Can you change this to the English version of this wikipedia page?

File: localflavor/mx/forms.py [hunk 2]
```diff
@@ -163,6 +163,50 @@ def _has_inconvenient_word(self, rfc):
...
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a valid CLABE.'),
+        'invalid_checksum': _('Invalid checksum for CLABE.'),
+    }
+
+    def __init__(self, min_length=18, max_length=18, *args, **kwargs):
+        clabe_re = r'^\d{18}$'
+        clabe_re = re.compile(clabe_re)
```
Comment (benkonrath) [inline localflavor/mx/forms.py]: The `RegexField` complies the regex so you don't need the `re.compile` line.

File: localflavor/mx/forms.py [hunk 3]
```diff
@@ -163,6 +163,50 @@ def _has_inconvenient_word(self, rfc):
...
+        super(MXCLABEField, self).__init__(clabe_re, min_length, max_length, *args, **kwargs)
+
+    def _checksum(self, value):
+        verification_digit = int(value[-1])
+        number = value[:-1]
+
+        ponderator = (3, 7, 1, 3, 7, 1, 3, 7, 1, 3, 7, 1, 3, 7, 1, 3, 7)
```
Comment (benkonrath) [inline localflavor/mx/forms.py]: It would be better to use `weights` or `weight_factors` for this variable instead of `ponderator`.

**Comment by benkonrath** [comment]
Thanks for this contribution! Just those small points to fix up. I was hoping to get this into the 1.3 release which I want to release 6 or 7 May.

It probably makes sense to also get the Spanish translations included with the release of this feature which means you'd need to add the translations in Transifex after this is merged. If you don't have access to Transifex , the timing won't work for the 1.3 release.

**Comment by agustin380** [comment]
@benkonrath Thanks fot he feedback! I made the changes you requested. Regarding translations, unfortunately I don't have access to Transifex, so it seems this PR will have to wait for the next release, unless something can be done about it...

**Comment by benkonrath** [comment]
Let's just push this off until the next release. I hope that's ok for you.

**Comment by benkonrath** [comment]
@agustin380 I just released 1.3. Can you change the versionadded to 1.4 and rebase so that I can merge this right away. Thanks.

**Comment by agustin380** [comment]
@benkonrath Done. I suppose translations will be taken care of later?

**Comment by benkonrath** [comment]
Thanks for correcting the date in the changelog. I'm not yet fully living in May it seems. :)

Merged via 4233c942c511b461d1893fec93aa9925f38a8cfd.

**Comment by benkonrath** [comment]
@agustin380 Translations are done with Transifex. You can signup and request access here:

https://www.transifex.com/django/django-localflavor/

You can ping claudep if you have an questions or run into trouble. Claude periodically syncs the translations from Transifex to this repo.

**Comment by agustin380** [comment]
@claudep Can you give me access to Transifex? I'd like to take care of the translations for this feature.

**Comment by claudep** [comment]
@agustin380 Did you request to be member of some team? Are you waiting for the coordinator approval? If yes, which team?

**Comment by agustin380** [comment]
@claudep I requested access to several Spanish dialects in https://www.transifex.com/django/django-localflavor/, I think they were the Argentina and Mexico ones (and neutral).

**Comment by claudep** [comment]
I added you to es ans es_MX. For es_AR, you should ping @ramiro who is coordinator.

**Comment by agustin380** [comment]
Thank you!

**Pull request closed without merging**